Inserter button in toolbar

hypest opened the pull request:
Addresses part of #58 

Adds a bottom toolbar button to trigger the Inserter. This is the minimum needed in order to be able to add a block when no block is selected or there are no blocks at all (e.g. when a new post, which is part of #182)

Changes in this PR:
1. Allow for empty string as initial data from the parent app
2. Inserter button in the bottom toolbar
3. Removal of inserter button from the inline toolbar

Scope not handled by this PR:
* Styling of the Inserter button to match https://github.com/wordpress-mobile/gutenberg-mobile/issues/58

![screenshot-1542108851184](https://user-images.githubusercontent.com/1032913/48410974-de6ac180-e748-11e8-8d89-3331e1036113.jpg)

To test 1:
* Run the app. No blocks are focused yet.
* Tap on the bottom toolbar Inserter button
* Select a block from the Inserter bottom sheet
* Scroll at the beginning of the content and notice the new block is there

To test 1:
* Run the app. No blocks are focused yet.
* Tap on the Image block to focus it
* Tap on the bottom toolbar Inserter button
* Select a block from the Inserter bottom sheet
* Notice the new block is inserted right after the Image block

marecar3 commented on src/block-management/block-toolbar.js:
```diff
@@ -1,11 +1,28 @@
+				<IconButton
+					className="block-list-appender__toggle"
+					label={ __( 'Add block' ) }
+					icon="insert"
```
The only concern here would be that we didn't share some constant with gutenberg project, so in the future, if someone changes those strings, it will break gutenberg-mobile.

marecar3 replied:
Sounds like a good plan :)

marecar3 commented on src/block-management/block-toolbar.js:
```diff
@@ -1,11 +1,28 @@
 		return (
 			<View style={ { height: 50, backgroundColor: '#DCDCDC', flexDirection: 'row' } }>
+				<IconButton
+					className="block-list-appender__toggle"
```
Not sure if we need className and what value it has for us?

hypest replied:
Ah, I left that in to tweak the button's styling later but, I shall remove it from this PR since the styling is not handled here.

hypest replied:
Done with e8df805.

etoledom commented:
One detail to notice:

In Gutenberg-Web, when I open an existing post with content, and add a new block with the `+` button at the top right corner, the new block is appended at the end of the document.

Should we try to copy that behavior in this PR or it is not important for now?
cc @hypest .

hypest commented:
> In Gutenberg-Web, when I open an existing post with content, and add a new block with the + button at the top right corner, the new block is appended at the end of the document.
> 
> Should we try to copy that behavior in this PR or it is not important for now?

Sounds reasonable, I'll add the logic, thanks @etoledom 👍

hypest commented:
Changed the code to add the new block at the end of the list with fb4dfda.
While at it, added a scroll animation when on Android. See d86dc09 and its commit comment.

hypest commented:
Addressed review comments, ready for another pass @marecar3 , @etoledom , thanks!

etoledom commented:
Looks great to me ✅ 

I envy the scroll to the new block on Android. Let's add a ticket to implement that on iOS :)

marecar3 commented:
Looks good to me ✅

hypest commented:
> I envy the scroll to the new block on Android. 

Heh, that's a happy accident that it's quick to do on the DataSource. I don't think that the current way of doing it will survive for long. We need to figure out the proper way to do UI adjustments like this (and anims).
